oak-run/src/test/java/org/apache/jackrabbit/oak/indexversion/PurgeOldIndexVersionIT.java

```diff
@@ -0,0 +1,129 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.indexversion;
+
+import ch.qos.logback.classic.Level;
+import org.apache.jackrabbit.oak.commons.junit.LogCustomizer;
+import org.apache.jackrabbit.oak.composite.blueGreen.IndexUtils;
+import org.apache.jackrabbit.oak.composite.blueGreen.Persistence;
+import org.apache.jackrabbit.oak.spi.state.NodeStore;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class PurgeOldIndexVersionIT {
+
+    private final Persistence.Config config = new Persistence.Config();
+
+    @Rule
+    public TemporaryFolder tempDir = new TemporaryFolder(new File("target"));
+
+    private File globalDir;
+    private File libsDir;
+    private File datastoreDir;
+    private File indexDir;
+    private LogCustomizer purgeOldIndexVersionLogger;
+
+    @Before
+    public void setup() {
+        purgeOldIndexVersionLogger = LogCustomizer.forLogger(PurgeOldIndexVersion.class.getName())
+                .enable(Level.INFO).contains("Found some index need to be purged over base").create();
+    }
+
+    @After
+    public void teardown() {
+        purgeOldIndexVersionLogger.finished();
+    }
+
+    @Test
+    public void test() throws Exception {
+        createFolders();
+        config.blobStore = Persistence.getFileBlobStore(datastoreDir);
+        config.indexDir = indexDir;
+        initGlobal();
+        initLibs();
+        compositeLibs();
+
+        purgeOldIndexVersionLogger.starting();
+        PurgeOldIndexVersion purgeOldIndexVersion = new PurgeOldIndexVersion();
+        Persistence p = Persistence.openComposite(globalDir, libsDir, config);
+        NodeStore n = p.getCompositeNodestore();
+        purgeOldIndexVersion.execute(n, true, 1, Arrays.asList("/oak:index"));
+
+        String purgeLog = "Found some index need to be purged over base'/oak:index/test':" +
+                " '[/oak:index/test-1 base=/oak:index/test versioned product=1 custom=0 operation:DELETE_HIDDEN_AND_DISABLE]'";
+        Assert.assertEquals(1, purgeOldIndexVersionLogger.getLogs().size());
+        Assert.assertEquals(purgeLog, purgeOldIndexVersionLogger.getLogs().get(0).toString());
```

Comment:
no need to call toString here
```suggestion
        Assert.assertEquals(purgeLog, purgeOldIndexVersionLogger.getLogs().get(0));
```